### ISSUE TITLE
Failed to upgrade the cluster because of image repository

### ISSUE THREAD

**Issue by yulinor** [labring/sealos#4618]
### Sealos Version

4.3.7 to 5.0 rc 4

### How to reproduce the bug?

I previously generated a Clusterfile, modified the svc pod network segment, node resource retention policy, closed kube-proxy, and ran a 1.27.7 cluster.
When using sealos v5.0rc4 to update the cluster, it failed because the foreign image could not be pulled.
Although you can pull the domestic image locally and modify the image name through tags.
But I can’t understand why the old cluster uses the image source sealos.hub, but when upgrading, kubeadm will use the original foreign image address.
Attempting to add imageRepository to the ClusterConfiguration fragment in .sealos/${clusterName}/Clusterfile to specify the image source failed to solve the problem. kubeadm fixedly selected foreign images.
After checking, there is no parameter to specify the image source in sealos run.
I think you can add this flag to specify the mirror source.

### What is the expected behavior?

Use the default `sealos.hub:5000` when upgrading

### What do you see instead?

kubeadm pulls foreign images by default

### Operating environment

```markdown
- Sealos version: 4.3.7 upgrade to 5.0.0 rc4 
- Kubernetes version: 1.27.7 to 1.28.8
- Operating system: openeuler 22.03
```

**Comment by bxy4543** [comment]
It may be that the upgrade function is incompatible because the kubeadm image repo addresses of versions 1.27 and 1.28 have changed.

**Comment by yangxggo** [comment]
fixed by #4721
